### ISSUE TITLE
Cherrypick libc++ fix for std::__debug_less in c++17

### ISSUE THREAD

**Issue by rupprecht** [llvm/llvm-project#53788]
This does not work with `-stdlib=libc++ -std=c++17 -D_LIBCPP_DEBUG=0`:

```c++
#include <algorithm>
constexpr int x() { return std::min<int>({1, 2, 3, 4}); }
static_assert(x() == 1);
```

This was broken in b07b5bd72716625e0976a84d23652d94d8d0165a (llvmorg-14-init-10193-gb07b5bd72716), and 99e5c5256ff2eeebd977f2c82567f52b1e534a1f fixes this.

**Comment by ldionne** [comment]
LGTM, I'll cherry-pick.

**Comment by tstellar** [comment]
Merged: c06cc1c3a7f8
